### DISCUSSION TITLE
Please help me: Error "Invalid call. Nonexistent function 'get_database_reference' in base 'Nil'."

### DISCUSSION THREAD

**Discussion by SuperCdat** [GodotNuts/GodotFirebase#430] [Q&A]
today, i'm learning to use realtime database in firebase, godot. but there's an error like this "Invalid call. Nonexistent function 'get_database_reference' in base 'Nil'.". Anyone know how to solve this please help me

**Comment by WolfgangSenff** [comment]
You have to get the reference first. Sounds like you did not read the wiki. Please check the wiki for how to use the realtime database.

**Reply by WolfgangSenff** [reply]
I just realized this page is written for 3.x. I'm not sure what version of Godot you're using. Let me know and if it's 4.x, I can update this page or make a new one accordingly: https://github.com/GodotNuts/GodotFirebase/wiki/Realtime-Database

**Reply by SuperCdat** [reply]
thank you for reply, i forgot this on github :D

**Comment by SuperCdat** [comment]
![image](https://github.com/user-attachments/assets/b642c8c9-3f8e-4a74-8777-c359c1e89581)
i am sure that i did every setup step by step, i filled all requirements in env file, set database rule to true, i gave up. Today i try it again but it make me disappointed again.

**Reply by WolfgangSenff** [reply]
I see. The error you're getting is because it's made outside of ready. Declare it and set it this way: 

``` 
var db_ref
func _ready():
  db_ref = what you had before
```

**Reply by WolfgangSenff** [reply]
Conversely, I believe you can declare the variable as you have it as onready and that should work.
